File: articles/status-quo-bias-institutional-inertia/fortran/status_quo_choice.f90
program status_quo_choice
  implicit none
  real :: value_status_quo, value_alternative, premium, switch_cost, loss_aversion, perceived_loss
  real :: utility_sq, utility_alt
  integer :: adoption

  value_status_quo = 0.50
  value_alternative = 0.68
  premium = 0.08
  switch_cost = 0.05
  loss_aversion = 1.50
  perceived_loss = 0.04

  utility_sq = value_status_quo + premium
  utility_alt = value_alternative - switch_cost - loss_aversion * perceived_loss

  if (utility_alt >= utility_sq) then
    adoption = 1
  else
    adoption = 0
  end if

  print *, "Synthetic alternative adoption under status quo bias:", adoption
end program status_quo_choice

File: articles/status-quo-bias-institutional-inertia/rust/status_quo_choice.rs
fn utility_status_quo(value_status_quo: f64, status_quo_premium: f64) -> f64 {
    value_status_quo + status_quo_premium
}

fn utility_alternative(value_alternative: f64, switch_cost: f64, loss_aversion: f64, perceived_loss: f64) -> f64 {
    value_alternative - switch_cost - loss_aversion * perceived_loss
}

fn choose_alternative(
    value_status_quo: f64,
    value_alternative: f64,
    premium: f64,
    switch_cost: f64,
    loss_aversion: f64,
    perceived_loss: f64,
) -> bool {
    utility_alternative(value_alternative, switch_cost, loss_aversion, perceived_loss)
        >= utility_status_quo(value_status_quo, premium)
}

fn main() {
    println!(
        "Synthetic alternative adoption under status quo bias: {}",
        choose_alternative(0.50, 0.68, 0.08, 0.05, 1.50, 0.04)
    );
}
